### ISSUE TITLE
[Rust] Implement new Concept Exercise: str-use

### ISSUE THREAD

**Issue by efx** [exercism/rust#1061]
This issue describes how to implement the `str-use` concept exercise for the rust track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

Introduce the student to the basics of working wih the `&str` type: string slices.

## Learning objectives

- Know how to work with `&str` and through the available methods
- Know that `&str` are a shared reference to a `String`
- Know that `&str` is immutable

## Out of scope

- the differences between `&str` and `String`
- `String`

## Concepts

- mutability
- `&str`

## Prerequisites

- slices
- basics

### Hints

- [string slices discussion in TRPL book][https://doc.rust-lang.org/book/ch04-03-slices.html#string-slices]
- [`&str` primitive documentation](https://doc.rust-lang.org/std/primitive.str.html)
- [`std::str`](https://doc.rust-lang.org/std/str/)

### After

- [string slices discussion in TRPL book][https://doc.rust-lang.org/book/ch04-03-slices.html#string-slices]
- [`&str` primitive documentation](https://doc.rust-lang.org/std/primitive.str.html)
- [`std::str`](https://doc.rust-lang.org/std/str/)

## Representer

- No changes needed

## Analyzer

- No changes needed

## Implementing

See our [implementation guide](https://github.com/exercism/v3/blob/master/languages/rust/howto/implementing-a-concept-exercise.md#how-to-implement-a-rust-concept-exercise) for guidance.

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.
